physics/eclipse_test.cpp

```diff
@@ -0,0 +1,167 @@
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include "physics/solar_system.hpp"
+#include "quantities/quantities.hpp"
+#include "quantities/si.hpp"
+#include "integrators/symplectic_runge_kutta_nyström_integrator.hpp"
+#include "geometry/epoch.hpp"
+#include "geometry/named_quantities.hpp"
+#include "quantities/elementary_functions.hpp"
+#include "geometry/grassmann.hpp"
+#include "testing_utilities/numerics.hpp"
+
+namespace principia {
+
+using astronomy::ICRFJ2000Equator;
+using geometry::JulianDate;
+using integrators::McLachlanAtela1992Order5Optimal;
+using quantities::si::Minute;
+using quantities::si::Metre;
+using quantities::si::Nano;
+using quantities::si::Milli;
+using quantities::si::Kilo;
+using quantities::ArcCos;
+using testing_utilities::AbsoluteError;
+using ::testing::Gt;
+using ::testing::Lt;
+using ::testing::AllOf;
+
+namespace physics {
+
+class EclipseTest : public testing::Test {
+ protected:
+  EclipseTest() {
+    solar_system_1950_.Initialize(
+        SOLUTION_DIR / "astronomy" / "gravity_model.proto.txt",
+        SOLUTION_DIR / "astronomy" /
+            "initial_state_jd_2433282_500000000.proto.txt");
+  }
+
+  SolarSystem<ICRFJ2000Equator> solar_system_1950_;
+};
+
+TEST_F(EclipseTest, Dummy) {
+  auto ephemeris = solar_system_1950_.MakeEphemeris(
+      McLachlanAtela1992Order5Optimal<Position<ICRFJ2000Equator>>(),
+      45 * Minute, 5 * Milli(Metre));
+
+  ephemeris->Prolong(JulianDate(2433374.5));  // Prolong until date of eclipse
+                                              // (Eclipse was 1950-04-02 but JD
+                                              // is 1950-04-03:00:00:00)
+  // pass body to Ephemeris.trajectory
```

Comment:
Comments should start with a capital letter and end with a period.  Also the wording is a bit odd confusing since it doesn't use `Ephemeris::trajectory`.  Maybe you meant something like "Extract the bodies from the ephemeris" or some such.